mobile-app/app/screens/TransactionAuthorization/PasscodePrompt.tsx

```diff
@@ -58,91 +59,126 @@ export function PasscodePrompt (props: PasscodePromptProps): JSX.Element {
+                    light={tailwind('text-gray-500')}
+                    style={tailwind('text-sm text-center')}
+                  >
+                    {props.transaction.description}
+                  </ThemedText>
+                )
+              }
+              </View>
+
+              {
+                props.status === 'PIN' && (
+                  <PinTextInput
+                    cellCount={props.pinLength}
+                    onChange={(pin) => {
+                      props.onPinInput(pin)
+                    }}
+                    testID='pin_authorize'
+                    value={props.pin}
+                  />
+                )
+              }
+
+              <Loading
+                message={props.status === 'SIGNING' ? props.loadingMessage : undefined}
+              />
+
+              {// upon retry: show remaining attempt allowed
+              (props.isRetry && props.attemptsRemaining !== undefined && props.attemptsRemaining !== props.maxPasscodeAttempt)
+                ? (
+                  <ThemedText
+                    dark={tailwind('text-darkerror-500')}
+                    light={tailwind('text-error-500')}
+                    style={tailwind('text-center text-sm font-bold mt-5')}
+                    testID='pin_attempt_error'
+                  >
+                    {translate('screens/PinConfirmation', `${props.attemptsRemaining === 1
+                      ? 'Last attempt or your wallet will be unlinked for your security'
+                      : 'Incorrect passcode. {{attemptsRemaining}} attempts remaining'}`, { attemptsRemaining: props.attemptsRemaining })}
+                  </ThemedText>
+                )
+                : null
+            }
+
+              {// on first time: warn user there were accumulated error attempt counter
+              (!props.isRetry && props.attemptsRemaining !== undefined && props.attemptsRemaining !== props.maxPasscodeAttempt)
+                ? (
+                  <ThemedText
+                    dark={tailwind('text-darkerror-500')}
+                    light={tailwind('text-error-500')}
+                    style={tailwind('text-center text-sm font-bold mt-5')}
+                    testID='pin_attempt_warning'
+                  >
+                    {translate('screens/PinConfirmation', `${props.attemptsRemaining === 1
+                      ? 'Last attempt or your wallet will be unlinked for your security'
+                      : '{{attemptsRemaining}} attempts remaining'}`, { attemptsRemaining: props.attemptsRemaining })}
+                  </ThemedText>
+                )
+                : null
+            }
+            </ThemedView>)}
       </ThemedView>
     </SafeAreaView>
   )
 }
 
+function SuccessMessage ({ message }: { message?: {title: string, description: string}}): JSX.Element | null {
+  if (message === undefined) {
+    return null
+  }
+
+  return (
+    <View style={tailwind('flex-col items-center p-6')}>
+      <ThemedIcon
```

Comment:
You can use this icon - https://icons.expo.fyi/MaterialCommunityIcons/check-circle-outline